### ISSUE TITLE
Need to update the `Document` interface

### ISSUE THREAD

**Issue by arronei** [patrickkettner/cookie-change-events#4]
You need to update the `Document` interface and update the way cookie works.

```
[Exposed=Window]
partial interface Document {
    attribute Cookie cookie;
}
```

**Comment by arronei** [comment]
Note: I am not sure how this is going to work for interop. You might need to do this...

`attribute (USVString or Cookie) cookie;`

**Comment by arronei** [comment]
```
dictionary CookieInit {
    USVString name;
    required USVString value;
    DOMString expirationDate;
};

[Exposed=Window]
[Constructor(CookieInit cookieInitDict)]
interface Cookie {
    readonly attribute USVString name;
    attribute USVString value;
    readonly attribute USVString domain;
    readonly attribute USVString path;
    readonly attribute boolean secure;
    readonly attribute boolean session;
    attribute DOMString expirationDate;
    stringifier USVString;
};

[Exposed=Window]
interface CookieJar{
    stringifier attribute USVString value;
    maplike<USVString, Cookie>;
};

partial interface Document {
    [PutForwards=value] attribute CookieJar cookie;
};
```

**Comment by arronei** [comment]
Another option for the maplike<USVString, Cookie>; in the CookieJar would be to use a setlike<Cookie>; this would give you the add() method you would rather have instead of the set() method.

**Comment by arronei** [comment]
I am thinking more about this and I think it would be great to be able to access the new Cookie object by the [] notation. You would not be able to set using that notation.

document.cookie["foo"]; would get you the full Cookie object for "foo".

you would have to add to CookieJar:
getter Cookie (USVString name);

**Comment by patrickkettner** [comment]
@arronei thinking about this a bit more after our chat, and the only way I would be comfortable with this would feel correct would be if I could do both

```js
document.cookie["foo"] = "bar"
```

and

```js
document.cookie["foo"].expirationDate = Date.now()
```

**Comment by arronei** [comment]
setter USVString (Cookie or USVString);

**Comment by patrickkettner** [comment]
This was fixed in cfad667
